lib/keyring_liberator.rb

```diff
@@ -21,12 +21,16 @@ def self.get_keyring(path)
     end
 
     def self.get_keyring_named(name)
-      get_all_keyrings.find { |k| k.name == name }
+      get_all_keyrings.find { |k| k.name == name && k.path == Dir.getwd }
```

Comment:
Sure, I was going down a different path just now, and moving that check into `preinstaller.rb`.  e.g.

```ruby
keyrings = KeyringLiberator.get_all_keyrings_named(project)
keyring = keyrings.find { |k| k.path == current_dir.to_s }
if keyrings.count > 1 && keyring.nil?
  ui.puts "Found multiple keyrings for project #{project.inspect}, but"
  ui.puts "no match found for current path (#{current_dir}):"
  keyrings.each do |found_keyring|
    ui.puts "- #{found_keyring.path}"
  end
  ui.puts "\nPress enter to create a new keyring, or `ctrl + c` to cancel"
  ui.gets
end
keyring ||= KeyringLiberator.get_keyring(current_dir)
```